weather_dl/download_pipeline/parsers.py

```diff
@@ -69,6 +69,86 @@ def date(candidate: str) -> datetime.date:
     return converted
 
 
+def time(candidate: str) -> datetime.time:
+    """Converts ECMWF-format time strings into a `datetime.time`.
+
+    Accepted time formats:
+    - HH:MM
+    - HHMM
+    - HH
+
+    For example:
+    - 18:00
+    - 1820
+    - 18
+
+    Note: If MM is omitted it defaults to 00.
+    """
+    converted = None
+
+    accepted_formats = ["%H", "%H:%M", "%H%M"]
+
+    for fmt in accepted_formats:
+        try:
+            converted = datetime.datetime.strptime(candidate, fmt).time()
+            break
+        except ValueError:
+            pass
+
+    if converted is None:
+        raise ValueError(
+            f"Not a valid time: '{candidate}'. Please use valid format."
+        )
+
+    return converted
+
+
+def day_month_year(key: str, candidate: t.Any) -> int:
+    """Converts day, month and year strings into 'int'.
+
+    Also validates the day and month input.
+    """
+    converted = None
+
+    def validate(key: str, candidate: int) -> None:
```

Comment:
This is a neat approach. However, I think it would be clearer / more succinct if you simply raised a `ValueError` if the candidate falls outside of the range.